gui/src/ts/components/benchmark/BenchmarkAnswerPage.tsx

```diff
@@ -1,5 +1,222 @@
 import * as React from "react";
+import {Frame} from "components/frame/Frame";
+import {useParams, Link} from "react-router-dom";
+import * as BenchmarkAnswerPageQueryDocument from "api/queries/benchmark/BenchmarkAnswerPageQuery.graphql";
+import {
+  BenchmarkAnswerPageQuery,
+  BenchmarkAnswerPageQueryVariables,
+  BenchmarkAnswerPageQuery_benchmarkById_datasetById_questionById_choices as QuestionAnswerChoice,
+} from "api/queries/benchmark/types/BenchmarkAnswerPageQuery";
+import {useQuery} from "@apollo/react-hooks";
+import * as _ from "lodash";
+import * as ReactLoader from "react-loader";
+import {ApolloException} from "@tetherless-world/twxplore-base";
+import {FatalErrorModal} from "components/error/FatalErrorModal";
+import {
+  Grid,
+  Breadcrumbs,
+  Typography,
+  Card,
+  CardContent,
+} from "@material-ui/core";
+import {Hrefs} from "Hrefs";
 
-export const BenchmarkAnswerPage: React.FunctionComponent = () => (
-  <div>Benchmark answer page</div>
+//localhost:9001/benchmark/benchmark0/dataset/benchmark0-test/submission/benchmark0-submission/question/benchmark0-test-0
+
+interface BenchmarkPathParams {
+  benchmarkId: string;
+  datasetId: string;
+  submissionId: string;
+  questionId: string;
+}
+
+const BenchmarkBreadcrumbs: React.FunctionComponent<Partial<
+  BenchmarkPathParams
+>> = ({benchmarkId, datasetId, submissionId, questionId}) => {
+  const breadcrumbsChildren: React.ReactNode = (() => {
+    const breadcrumbs: React.ReactNodeArray = [
+      <Link to={Hrefs.benchmarks}>Benchmarks</Link>,
+    ];
+
+    if (!benchmarkId) {
+      return breadcrumbs;
+    }
+
+    const benchmark = Hrefs.benchmark({id: benchmarkId});
+    breadcrumbs.push(<Link to={benchmark.home}>{benchmarkId}</Link>);
+
+    if (!datasetId) {
+      return breadcrumbs;
+    }
+
+    breadcrumbs.push(<Typography>Datasets</Typography>);
+
+    const dataset = benchmark.dataset({id: datasetId});
+    breadcrumbs.push(<Link to={dataset.home}>{datasetId}</Link>);
+
+    if (!submissionId) {
+      return breadcrumbs;
+    }
+
+    breadcrumbs.push(<Typography>Submissions</Typography>);
+
+    const submission = dataset.submission({id: submissionId});
+    breadcrumbs.push(<Link to={submission.home}>{submissionId}</Link>);
+
+    if (!questionId) {
+      return breadcrumbs;
+    }
+
+    breadcrumbs.push(<Typography>Questions</Typography>);
+
+    breadcrumbs.push(
+      <Link to={submission.question({id: questionId})}>{questionId}</Link>
+    );
+
+    return breadcrumbs;
+  })();
+
+  return <Breadcrumbs>{breadcrumbsChildren}</Breadcrumbs>;
+};
+
+const NotFound: React.FunctionComponent<{label: string}> = ({label}) => {
```

Comment:
This is something reusable, too. I think we already do have something like it in NodePage.tsx? Please factor into a component in error/.